### PR TITLE
Reset view when idle and add clear cart

### PR DESCRIPTION
**Pull request by patkan**
This patch is #41, but refactored.
Still any issues?

### CODE REVIEW

**Comment by mgmax** [comment]
don't merge before #45 is fixed

**Comment by patkan** [comment]
Watch out! I rebased this branch!

File: FabLabKasse/shopping/backend/oerp.py [hunk 1]
```diff
@@ -232,6 +232,9 @@ def delete_order_line(self, order_line_id):
         self.oerp.unlink('sale.order.line', [order_line_id])
 
     def delete_current_order(self):
+        """ deletes current order and calls self.set_current_order(None)
+        :return:
+        """
```
Comment (mgmax) [inline FabLabKasse/shopping/backend/oerp.py]: - [x]  das erbt den docstring der oberklasse, wär also besser den wegzulassen

File: FabLabKasse/shopping/backend/offline_base.py [hunk 2]
```diff
@@ -284,8 +284,10 @@ def create_order(self):
         return id(new_order)
 
     def delete_current_order(self):
+        """ deletes current order and calls self.set_current_order(None)
+        :return:
+        """
```
Comment (mgmax) [inline FabLabKasse/shopping/backend/offline_base.py]: - [x] ebenso: das erbt den docstring der oberklasse, wär also besser den wegzulassen

File: FabLabKasse/gui.py [hunk 3]
```diff
@@ -817,8 +835,50 @@ def leaveSearch(self, keepResultsVisible=False):
...
+        idle_keyword = "idle"
+        if idle_state[0] == idle_keyword:
+            return True
+        elif idle_state[0] is None and self.idleTracker.last_state == idle_keyword:
+            return True
+        else:
+            return False
 
+    def _reset_if_idle(self):
+        """resets the category-view of the GUI if it is idle for a certain timespan"""
```
Comment (mgmax) [inline FabLabKasse/gui.py]: - [x]  +doc:
  This function might be called even if another dialog like payup is currently active. Therefore the current order is not modified or updated, as result might be undefined during payup.

**Comment by mgmax** [comment]
For me it only works with "libXss.so.1", otherwise the lib cannot be found.
Would be great if you looked at my 3 comments regarding documentation.

except for that: Super, nice code, ready to merge :)

**Pull request merged**